### ISSUE TITLE
Unable to access learn.test.up2niversity.eu

### ISSUE THREAD

**Issue by ebocchi** [up2university/docker-moodle#39]
I am unable to log in to the test Moodle instance.
Could it be due to the externalization of Moodle configuration? 
https://github.com/up2university/docker-moodle/issues/31

Upon clicking on Log In (top right), I get an unhandled exception:
![screenshot from 2018-02-20 09-58-13](https://user-images.githubusercontent.com/16349448/36415035-bbe7d90e-1624-11e8-977e-068cf24b182c.png)

**Comment by michzimny** [comment]
The same happens for me.

>  Could it be due to the externalization of Moodle configuration?

No, externalization is not yet present at this instance.

**Comment by athird** [comment]
@fubarable? I suspect this relates to the same issue that @nadavkav has been having too. [SSO issues](https://github.com/up2university/single-sign-on/issues)

**Comment by michzimny** [comment]
@athird, you meant "foobarable" but he's not yet added to the project (an invitation awaits).

The same happens in learn2.test, however in production it works fine.

**Comment by athird** [comment]
Ah, I did, thanks! 

Strange, I haven't changed anything in the SSO configuration on either of those since we enabled it, and it worked then.

**Comment by michzimny** [comment]
Yesterday I merged https://github.com/up2university/docker-moodle/pull/37 what causes deployments to those two instances but it seems rather unrelated.

**Comment by athird** [comment]
I can't see any differences in the configurations.

**Comment by nadavkav** [comment]
Yep. have same issues.
Yesterday and today, I ran some test with our IUCC IDP team and it seems some of the attributes are not passed with the certificate. with the new authentication page. but if we used the old one it works fine. and also when we use the up2university.eu (wordpress) we are able to login successfully.

I have installed https://chrome.google.com/webstore/detail/saml-chrome-panel/paijfdbeoenhembfhkhllainmocckace
And I can send the debug errors, if it can help.

**Comment by foobarable** [comment]
Hi, ill check the logs.

**Comment by foobarable** [comment]
I haven't changed anything as well, though. Could you give me the exact time you tried?

**Comment by foobarable** [comment]
@athird @michzimny Did the certificate of the moodle instance maybe change? Its really not a good idea to use letsencrypt certificates for encrypting the SAML messages. You can use self signed and more persistent certificates though.

**Comment by michzimny** [comment]
> Could you give me the exact time you tried?

@foobarable, it happens all the time. Please go to https://learn.test.up2university.eu/ and click "Log in" in the top right corner.

> Did the certificate of the moodle instance maybe change? 

The last cert renewal took place on 29th January.

It's more complex issue about letsencrypt certs. I send you an email.

**Comment by foobarable** [comment]
Problem is that simplesamlphp can't decrypt the SAML message it is getting and thus fails to auth. In my experience it's 99% a certificate error in the metadata. (Mismatch between actual cert and cert from metadata)

**Comment by michzimny** [comment]
Ok, thanks. So @athird needs to check the cert used for SAML encryption.

**Comment by athird** [comment]
Yup, just been looking. I've no idea what did this, it *looks* like someone clicked the Regenerate Certificate button in the Moodle admin interface. I can't restore the previous cert. as it seems the private key was overwritten with the new one. 

@foobarable Can you update the metadata on your side to use the new ones, will that fix it? If it does/will, then I'll lock the cert. files on Moodle to try to stop this happening again.

**Comment by michzimny** [comment]
I can restore something from backups if needed. But I'm not whether it can be easily extracted from a backup.

**Comment by athird** [comment]
They're in /var/moodledata/saml2, if that helps - I think that's kept externally to the container, right?

**Comment by athird** [comment]
@michzimny Thanks, yes, it does - I've copied those files back in and it's taking me to the SSO login page now. Can you 9and anyone else following this thread) check it from your sides just to make sure, please?

Thanks,

Allan

**Comment by ebocchi** [comment]
Hi @athird 
I get the SSO login page but no luck when I get redirected back to moodle:
![screenshot from 2018-02-21 12-32-11](https://user-images.githubusercontent.com/16349448/36477960-4fb424ec-1703-11e8-8a0e-40b81acda940.png)

It happens both with EduGain and Google.

**Comment by michzimny** [comment]
I get something similiar.

**Comment by foobarable** [comment]
Yepp, same here. Are you sure you didn't change anything at the SP configuration?

**Comment by athird** [comment]
I'm sure, and I just checked and can't see anything. I'm going to email you (@foobarable) some log output.

**Comment by foobarable** [comment]
I already turned on the debug mode on sso to see the SAML messages. So far nothing really suspicious

**Comment by foobarable** [comment]
We have a patch for the issue but waiting for our test instance to be avaibable in the edugain AAI so we can test if it fixes the issue.

**Comment by foobarable** [comment]
Issue was solved by removing the EPTID from the SAML response.

**Comment by michzimny** [comment]
The problem seems to be solved for learn.test.up2university.eu

**Comment by nadavkav** [comment]
Indeed, I do not see that problem, but I still can not login :disappointed:

**Comment by foobarable** [comment]
Why can't you login?

**Comment by nadavkav** [comment]
Here is a screen capture clip of what I am experiencing for two weeks now:
https://youtu.be/gOdXQcerTZQ

**Comment by foobarable** [comment]
```Mar  2 12:12:20 ssoUp2u simplesamlphp[43840]: 7 [bc4259b8b9]       <saml:Attribute Name="urn:oid:1.3.6.1.4.1.5923.1.1.1.10" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:uri">
Mar  2 12:12:20 ssoUp2u simplesamlphp[43840]: 7 [bc4259b8b9]         <saml:AttributeValue xsi:type="xs:string">youruniqueid</saml:AttributeValue>
Mar  2 12:12:20 ssoUp2u simplesamlphp[43840]: 7 [bc4259b8b9]       </saml:Attribute>
Mar  2 12:12:20 ssoUp2u simplesamlphp[43840]: 3 [bc4259b8b9] SimpleSAML_Error_Error: UNHANDLEDEXCEPTION

Mar  2 12:12:20 ssoUp2u simplesamlphp[43840]: 3 [bc4259b8b9] Caused by: SAML2\Exception\RuntimeException: A "urn:oid:1.3.6.1.4.1.5923.1.1.1.10" (EPTI) attribute value must be a NameID, none found for value no. "0"

Your IDP does the same mistake as we did, it sent the EPTID as string and not as name ID.

**Comment by nadavkav** [comment]
Thank you @foobarable , I will look into it with our IT the maintain the IDP

**Comment by foobarable** [comment]
Maybe I should implement a filter for it to throw the attribute away if its not in the valid form.

**Comment by nadavkav** [comment]
That could be useful. can you?

**Comment by foobarable** [comment]
Yes, will try it in the test system. Don't want to do it on the production instance. Nevertheless, please check the attribute from your side as well.

**Comment by nadavkav** [comment]
I already sent an email to our IT about it, but we are already on the weekend and I will get an answer on sunday afternoon, the earliest.

**Comment by orenshani7** [comment]
Hi, I am the IT guy...

@foobarable , can you send me the entire SAML correspondence as it looks from your side? Please send to oshani@iucc.ac.il

Thanks,

Oren

**Comment by foobarable** [comment]
[ x ] Done

**Comment by nadavkav** [comment]
![snapshot228](https://user-images.githubusercontent.com/231168/36975038-7b07a8ee-2081-11e8-8ad4-256a2be0fb17.png)

**Comment by foobarable** [comment]
I just meant I sent @orenshani7 the SAML message

**Comment by michzimny** [comment]
Is it still a problem?

**Comment by michzimny** [comment]
@nadavkav, is it still a problem?

**Comment by nadavkav** [comment]
Nope. works fine :smile: thank you. (sorry for not reporting on this earlier)
